tests/framework/extensions/nodes/ec2/ec2.go

```diff
@@ -184,3 +185,47 @@ func getSSHKeyName(sshKeyName string) string {
 	stringSlice := strings.Split(sshKeyName, ".")
 	return stringSlice[0]
 }
+
+// IsNodeDeleted checks if the node is in terminated state or does not exist.
+// it throws an error if multiple nodes with the same name exist.
+func IsNodeDeleted(client *rancher.Client, nodeName string) (bool, error) {
```

Comment:
Also we're looking to deprecate the ec2 client, so I rather not add more things to it.

Reply:
Agreed